### PR TITLE
feat: Adding integration

### PR DESCRIPTION
**Pull request by ronh1**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress

## Description
Adding the enhancement for Rapid7 Insight IDR integration.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Review by JudahSchwartz — COMMENTED** [review]
fantastic!

File: Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR_image.png
[the remark this answers is not in the document]
Comment (ronh1) [reply Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR_image.png:1]: It wasn't in the XSOAR best practice size :)

File: Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR_test.py [hunk 1]
```diff
@@ -375,26 +450,19 @@ def test_insight_idr_download_logs(requests_mock) -> None:
...
-        },
-        proxy=False
+        f"https://{REGION}.api.insight.rapid7.com/log_search/download/logs/x:y", text=mock_response
     )
-    response = insight_idr_download_logs_command(client, 'x:y')
 
-    assert (response.get('File', '')[-4:]) == '.log'
+    response = insight_idr_download_logs_command(mock_client, "x:y")
 
+    assert (response.get("File", "")[-4:]) == ".log"
```
[the remark this answers is not in the document]
Comment (ronh1) [reply Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR_test.py:462]: I don't care to touch it, but this code is relevant for the existing integration. What do you think?

File: Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.py [hunk 2]
```diff
@@ -1,104 +1,574 @@
...
+            max_investigations_to_close (int, optional): The maximum number of alerts to close. Defaults to None.
+
+        Returns:
+            dict: API response from Insight IDR API.
+        """
+        body = remove_empty_elements(
+            {
+                "source": source,
+                "alert_type": alert_type,
+                "disposition": disposition.replace(" ", "_") if disposition else None,
```
[the remark this answers is not in the document]
Comment (ronh1) [reply Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.py:211]: We decided to use without _ in the design part, to facilitate the user.

File: Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.py [hunk 3]
```diff
@@ -722,20 +1483,21 @@ def fetch_incidents(client: Client,
...
-                                        settings={'RETURN_AS_TIMEZONE_AWARE': False})
+    investigations = client.list_investigations(
+        start_time=last_fetch.strftime(DATE_FORMAT), size=size
+    )
+    for investigation in investigations.get("data", []):
+        investigation_created_time = investigation.get("created_time")
+        created_time = dateparser.parse(
+            investigation_created_time, settings={"RETURN_AS_TIMEZONE_AWARE": False}
+        )
         assert created_time is not None, f"could not parse {investigation_created_time}"
```
[the remark this answers is not in the document]
Comment (ronh1) [reply Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.py:1503]: I don't care to touch it, but this code is relevant for the existing integration. What do you think?

File: Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml [hunk 4]
```diff
@@ -1,435 +1,1066 @@
...
+          required: true
+        - description: "The optional 0 based index of the page to retrieve. Must be an integer greater than or equal to 0."
+          name: index
+          defaultValue: "0"
+        - description: "The optional size of the page to retrieve. Must be an integer greater than 0 or less than or equal to 1000."
+          name: page_size
+        - description: The maximum number of records to retrieve.
+          name: limit
+          defaultValue: "50"
+        - description: A comma-separated list of investigation statuses to include in the result. For example,Open,Closed.
```
Comment (ShirleyDenkberg) [inline Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml:99]:
```suggestion
        - description: A comma-separated list of investigation statuses to include in the result. For example, Open,Closed.
```

File: Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml [hunk 5]
```diff
@@ -1,435 +1,1066 @@
...
+          name: statuses
+          auto: PREDEFINED
+          predefined:
+            - open
+            - investigating
+            - closed
+          isArray: true
+        - description: A comma-separated list of investigation sources to include in the result. For example,User,Alert. Relevant when api_version is V2 only.
```
Comment (ShirleyDenkberg) [inline Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml:107]:
```suggestion
        - description: A comma-separated list of investigation sources to include in the result. For example, User,Alert. Relevant when api_version is V2 only.
```

File: Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml [hunk 6]
```diff
@@ -1,435 +1,1066 @@
...
+          name: sources
+          auto: PREDEFINED
+          predefined:
+            - User
+            - Alert
+          isArray: true
+        - name: priorities
+          description: A comma-separated list of investigation priorities to include in the result. For example,Low,Medium. Relevant when api_version is V2 only.
```
Comment (ShirleyDenkberg) [inline Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml:115]:
```suggestion
          description: A comma-separated list of investigation priorities to include in the result. For example, Low,Medium. Relevant when api_version is V2 only.
```

File: Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml [hunk 7]
```diff
@@ -1,435 +1,1066 @@
...
+          auto: PREDEFINED
+          predefined:
+            - Unspecified
+            - Low
+            - Mediun
```
Comment (ShirleyDenkberg) [inline Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml:120]:
```suggestion
            - Medium
```

File: Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml [hunk 8]
```diff
@@ -1,435 +1,1066 @@
...
+            - High
+            - Critical
+          isArray: true
+        - name: assignee_email
+          description: A user's email address. Only investigations assigned to that user will be included. For example,test@test.com.
```
Comment (ShirleyDenkberg) [inline Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml:125]:
```suggestion
          description: A user's email address. Only investigations assigned to that user will be included. For example, test@test.com.
```

File: Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml [hunk 9]
```diff
@@ -1,435 +1,1066 @@
...
+      name: rapid7-insight-idr-download-logs
+      outputs: []
+    - arguments:
+        - description: Logentries log key.
+          name: log_id
+          required: true
+        - description: "A valid LEQL query to run against the logmore info: https://docs.rapid7.com/insightidr/build-a-query/"
+          name: query
+          required: true
+        - description: A time range string (i.e 1 week, 1 day) - While using this parameter, start_time and end_time isn't needed.
```
Comment (ShirleyDenkberg) [inline Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml:583]:
```suggestion
        - description: A time range string (i.e., 1 week, 1 day). When using this parameter, start_time and end_time isn't needed.
```

File: Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml [hunk 10]
```diff
@@ -1,435 +1,1066 @@
...
+            - Unspecified
+            - Low
+            - Mediun
+            - High
+            - Critical
+          isArray: true
+        - name: assignee_email
+          description: A user's email address. Only investigations assigned to that user will be included. For example,test@test.com.
+        - name: time_range
+          description: An optional time range string (i.e 1 week, 1 day).
```
Comment (ShirleyDenkberg) [inline Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml:127]:
```suggestion
          description: An optional time range string (i.e., 1 week, 1 day).
```

File: Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml [hunk 11]
```diff
@@ -1,435 +1,1066 @@
...
+          description: How this investigation was generated.
+          type: String
+        - contextPath: Rapid7InsightIDR.Investigation.title
+          description: The name of the investigation.
+          type: String
+        - contextPath: Rapid7InsightIDR.Investigation.organization_id
+          description: The ID of the organization that owns this investigation. Relevant when api_version is V2 only.
+          type: String
+        - contextPath: Rapid7InsightIDR.Investigation.rrn
+          description: The RRN of the investigation. Relevant when api_version is V2 only.
```
Comment (ShirleyDenkberg) [inline Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml:194]:
```suggestion
          description: The Rapid7 Resource Names of the investigation. Relevant when api_version is V2 only.
```

File: Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml [hunk 12]
```diff
@@ -1,435 +1,1066 @@
...
+          type: String
+        - contextPath: Rapid7InsightIDR.Investigation.id
+          description: The ID of the investigation. Relevant when api_version is V1 only.
+          type: String
+        - contextPath: Rapid7InsightIDR.Investigation.alert.type
+          description: Type of alert in the investigation. Relevant when api_version is V1 only.
+          type: String
+        - contextPath: Rapid7InsightIDR.Investigation.alert.type_description
+          description: Type description of alert in the investigation. Relevant when api_version is V1 only.
```
Comment (ShirleyDenkberg) [inline Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml:203]:
```suggestion
          description: Type description of the type of alert in the investigation. Relevant when api_version is V1 only.
```

File: Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml [hunk 13]
```diff
@@ -1,435 +1,1066 @@
...
+          type: String
+        - contextPath: Rapid7InsightIDR.Investigation.alert.first_event_time
+          description: First event time of alert in the investigation. Relevant when api_version is V1 only.
```
Comment (ShirleyDenkberg) [inline Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml:206]:
```suggestion
          description: First event time of the alert in the investigation. Relevant when api_version is V1 only.
```

File: Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml [hunk 14]
```diff
@@ -1,435 +1,1066 @@
...
+          defaultValue: V1
+          auto: PREDEFINED
+          predefined:
+            - V1
+            - V2
+        - name: investigation_id
+          description: The ID or RRN (If api_version=V2, the ID of the investigation must be in the RRN format) of the investigation to retrieve. Use rapid7-insight-idr-list-investigations to retrieve all investigation IDs.
+          required: true
+          default: true
+      description: Get a specific investigation. This investigation is specified by either ID or RRN (If multi-customer set to true, the investigation_id must be in the RRN format).
```
Comment (ShirleyDenkberg) [inline Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml:220]:
```suggestion
      description: Get a specific investigation. This investigation is specified by either ID or apid7 Resource Names (RRN). (If multi-customer is set to true, the investigation_id must be in the RRN format).
```

File: Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml [hunk 15]
```diff
@@ -1,435 +1,1066 @@
...
+          description: How this investigation was generated.
+          type: String
+        - contextPath: Rapid7InsightIDR.Investigation.title
+          description: The name of the investigation.
+          type: String
+        - contextPath: Rapid7InsightIDR.Investigation.organization_id
+          description: The ID of the organization that owns this investigation. Relevant when api_version is V2 only.
+          type: String
+        - contextPath: Rapid7InsightIDR.Investigation.rrn
+          description: The RRN of the investigation. Relevant when api_version is V2 only.
```
Comment (ShirleyDenkberg) [inline Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml:263]:
```suggestion
          description: The Rapid7 Resource Names of the investigation. Relevant when api_version is V2 only.
```

File: Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml [hunk 16]
```diff
@@ -1,435 +1,1066 @@
...
+          type: String
+        - contextPath: Rapid7InsightIDR.Investigation.id
+          description: The ID of the investigation. Relevant when api_version is V1 only.
+          type: String
+        - contextPath: Rapid7InsightIDR.Investigation.alert.type
+          description: Type of alert in the investigation. Relevant when api_version is V1 only.
+          type: String
+        - contextPath: Rapid7InsightIDR.Investigation.alert.type_description
+          description: Type description of alert in the investigation. Relevant when api_version is V1 only.
```
Comment (ShirleyDenkberg) [inline Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml:272]:
```suggestion
          description: The description of the alert type in the investigation. Relevant when api_version is V1 only.
```

File: Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml [hunk 17]
```diff
@@ -1,435 +1,1066 @@
...
+          name: disposition
+          defaultValue: Not Applicable
+          auto: PREDEFINED
+          predefined:
+            - Undecided
+            - Benign
+            - Malicious
+            - Not Applicable
+        - name: detection_rule_rrn
+          description: The RRN of the detection rule. Only investigations that are associated with this detection rule will be closed. If a detection rule RRN is given, thealert_typeis required to be 'Attacker Behavior Detected'.  Userapid7-insight-idr-get-investigationto retrieve the investigationdetection_rule_rrn.
```
Comment (ShirleyDenkberg) [inline Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml:304]:
```suggestion
          description: The Rapid7 Resource Names (RRN) of the detection rule. Only investigations that are associated with this detection rule will be closed. If a detection rule RRN is given, thealert_typeis required to be 'Attacker Behavior Detected'.  Userapid7-insight-idr-get-investigationto retrieve the investigationdetection_rule_rrn.
```

File: Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml [hunk 18]
```diff
@@ -1,435 +1,1066 @@
...
+          predefined:
+            - V1
+            - V2
+        - description: Comma-separated list of the ID or RRN (If api_version=V2, the ID of the investigation must be in the RRN format) of the investigation to assign the user to. Use rapid7-insight-idr-list-investigations to retrieve all investigation IDs.
+          name: investigation_id
+          required: true
+          default: true
+        - name: user_email_address
+          required: true
+          description: The email address of the user to assign to this Investigation. Same email used to log into the insight platform. For example, test@test.com.
```
Comment (ShirleyDenkberg) [inline Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml:323]:
```suggestion
          description: The email address of the user to assign to this investigation. This is the same email used to log into the insight platform. For example, test@test.com. 
```

File: Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml [hunk 19]
```diff
@@ -1,435 +1,1066 @@
...
+          description: How this investigation was generated.
+          type: String
+        - contextPath: Rapid7InsightIDR.Investigation.title
+          description: The name of the investigation.
+          type: String
+        - contextPath: Rapid7InsightIDR.Investigation.organization_id
+          description: The ID of the organization that owns this investigation. Relevant when api_version is V2 only.
+          type: String
+        - contextPath: Rapid7InsightIDR.Investigation.rrn
+          description: The RRN of the investigation. Relevant when api_version is V2 only.
```
Comment (ShirleyDenkberg) [inline Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml:364]:
```suggestion
          description: The Rapid7 Resource Names of the investigation. Relevant when api_version is V2 only.
```

File: Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml [hunk 20]
```diff
@@ -1,435 +1,1066 @@
...
```
Comment (ShirleyDenkberg) [inline Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml:460]:
```suggestion
          description: The Rapid7 Resource Names of the investigation. Relevant when api_version is V2 only.
```

File: Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml [hunk 21]
```diff
@@ -1,435 +1,1066 @@
...
+        - description: 'The maximum number of log events to download; cannot exceed 20 million. The default is 20 million. The argument value should be written like this: "10 thousand" or "2 million").'
+          name: limit
+      description: Downloads logs for from your InsightsIDR instance. The maximum number of logs per call is 10.
+      name: rapid7-insight-idr-download-logs
+      outputs: []
+    - arguments:
+        - description: Logentries log key.
+          name: log_id
+          required: true
+        - description: "A valid LEQL query to run against the logmore info: https://docs.rapid7.com/insightidr/build-a-query/"
```
Comment (ShirleyDenkberg) [inline Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml:580]:
```suggestion
        - description: "A valid LEQL query to run against the log. For more information: https://docs.rapid7.com/insightidr/build-a-query/"
```

File: Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml [hunk 22]
```diff
@@ -1,435 +1,1066 @@
...
+          type: String
+        - contextPath: Rapid7InsightIDR.Event.timestamp
+          description: Time when the event was triggered.
+          type: Number
+    - arguments:
+        - description: ID of the log set.
+          isArray: true
+          name: log_set_id
+          required: true
+        - description: "A valid LEQL query to run against the logmore info: https://docs.rapid7.com/insightidr/build-a-query/"
```
Comment (ShirleyDenkberg) [inline Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml:610]:
```suggestion
        - description: "A valid LEQL query to run against the log. For more information: https://docs.rapid7.com/insightidr/build-a-query/"
```

File: Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml [hunk 23]
```diff
@@ -1,435 +1,1066 @@
...
+          isArray: true
+          name: log_ids
+          required: true
+        - description: "Lower bound of the time range you want to query against. Format: UNIX timestamp in milliseconds. This is optional if time_range is supplied."
+          name: start_time
+        - description: "Upper bound of the time range you want to query against. Format: UNIX timestamp in milliseconds."
+          name: end_time
+        - description: 'The relative time range in a readable format. Optional if "from" \ is supplied. For example: Last 4 Days. Note that if start_time, end_time and\ \ time_range is not provided the default will be Last 3 days.'
+          name: time_range
+        - description: "The LEQL query to match desired log events. Do not use a calculation.more info: https://docs.rapid7.com/insightidr/build-a-query/"
```
Comment (ShirleyDenkberg) [inline Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml:569]:
```suggestion
        - description: "The LEQL query to match desired log events. Do not use a calculation. For more information: https://docs.rapid7.com/insightidr/build-a-query/"
```

File: Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml [hunk 24]
```diff
@@ -1,435 +1,1066 @@
...
+          description: The disposition for the investigation. Select a disposition to indicate whether the investigation represented a legitimate threat.
+          defaultValue: Undecided
+          auto: PREDEFINED
+          predefined:
+            - Undecided
+            - Benign
+            - Malicious
+            - Not Applicable
+        - name: user_email_address
+          description: The email address of the user to assign to this Investigation. Same email used to log into the insight platform. Use rapid7-insight-idr-list-users to retrieve the user email list.
```
Comment (ShirleyDenkberg) [inline Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml:669]:
```suggestion
          description: The email address of the user to assign to this investigation. This is the same email used to log into the insight platform. Use rapid7-insight-idr-list-users to retrieve the user email list.
```

File: Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml [hunk 25]
```diff
@@ -1,435 +1,1066 @@
...
+        - name: disposition
+          description: The disposition for the investigation. Select a disposition to indicate whether the investigation represented a legitimate threat.
+          auto: PREDEFINED
+          predefined:
+            - Undecided
+            - Benign
+            - Malicious
+            - Not Applicable
+        - name: user_email_address
+          description: The email address of the user to assign to this Investigation. Same email used to log into the insight platform.
```
Comment (ShirleyDenkberg) [inline Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml:747]:
```suggestion
          description: The email address of the user to assign to this investigation. This is the same email used to log into the insight platform.
```

File: Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml [hunk 26]
```diff
@@ -1,435 +1,1066 @@
...
+          description: How this investigation was generated.
+          type: String
+        - contextPath: Rapid7InsightIDR.Investigation.title
+          description: The name of the investigation.
+          type: String
+        - contextPath: Rapid7InsightIDR.Investigation.organization_id
+          description: The ID of the organization that owns this investigation.
+          type: String
+        - contextPath: Rapid7InsightIDR.Investigation.rrn
+          description: The RRN of the investigation.
```
Comment (ShirleyDenkberg) [inline Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml:803]:
```suggestion
          description: The Rapid7 Resource Names of the investigation.
```

File: Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml [hunk 27]
```diff
@@ -1,435 +1,1066 @@
...
```
Comment (ShirleyDenkberg) [inline Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml:912]:
```suggestion
          description: The Rapid7 Resource Names of the investigation.
```

File: Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml [hunk 28]
```diff
@@ -1,435 +1,1066 @@
...
+          type: String
+    - name: rapid7-insight-idr-update-investigation
+      description: Updates multiple fields in a single operation for an investigation, specified by ID or RRN (If multi-customer set to true, the investigation_id must be in the RRN format). Use rapid7-insight-idr-list-investigations to retrieve all investigation IDs.
```
Comment (ShirleyDenkberg) [inline Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml:714]:
```suggestion
      description: Updates multiple fields in a single operation for an investigation, specified by ID or Rapid7 Resource Names (RRN). (If multi-customer set to true, the investigation_id must be in the RRN format). Use rapid7-insight-idr-list-investigations to retrieve all investigation IDs.
```

File: Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml [hunk 29]
```diff
@@ -1,435 +1,1066 @@
...
+      arguments:
+        - name: investigation_id
+          description: The ID or RRN (If api_version=V2, the ID of the investigation must be in the RRN format) of the investigation to to update.
```
Comment (ShirleyDenkberg) [inline Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml:717]:
```suggestion
          description: The ID or Rapid7 Resource Names (RRN) of the investigation to to update. (If api_version=V2, the ID of the investigation must be in the RRN format).
```

File: Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml [hunk 30]
```diff
@@ -1,435 +1,1066 @@
...
+        - name: rrn
+          description: The RRN (unique identifier for user.) of the user to retrieve. When using this argument, all the other irelevant.
+        - name: first_name
+          description: Filter investigations by comma-separated values to include only those equals the specified values in the 'first_name' field. Choose search operator to define the operator type.
+          isArray: true
+        - name: last_name
+          description: Filter investigations by comma-separated values to include only those equals the specified values in the 'last_name' field. Choose search operator to define the operator type.
+          isArray: true
+        - name: name
+          description: Filter investigations by comma-separated values to include only those equals the specified values in the 'name' field. Choose search operator to define the operator type.
```
Comment (ShirleyDenkberg) [inline Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml:1010]:
```suggestion
          description: A comma-separated list of first names. Choose search operator to define the operator type.
          isArray: true
        - name: last_name
          description: A comma-separated list of last names. Choose search operator to define the operator type.
          isArray: true
        - name: name
          description: A comma-separated list of names. Choose search operator to define the operator type.
```

File: Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml [hunk 31]
```diff
@@ -1,435 +1,1066 @@
...
+          description: The name of the investigation.
+          type: String
+        - contextPath: Rapid7InsightIDR.Investigation.organization_id
+          description: The ID of the organization that owns this investigation.
+          type: String
+        - contextPath: Rapid7InsightIDR.Investigation.rrn
+          description: The RRN of the investigation.
+          type: String
+    - name: rapid7-insight-idr-list-investigation-alerts
+      description: Retrieve and list all alerts associated with an investigation, with the given ID or RRN. The listed alerts are sorted in descending order by alert create time. Use rapid7-insight-idr-list-investigations to retrieve all investigation IDs.
```
Comment (ShirleyDenkberg) [inline Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.yml:915]:
```suggestion
      description: Retrieve and list all alerts associated with an investigation, with the given ID or Rapid7 Resource Names (RRN). The listed alerts are sorted in descending order by alert create time. Use rapid7-insight-idr-list-investigations to retrieve all investigation IDs.
```

File: Packs/Rapid7_InsightIDR/ReleaseNotes/2_0_0.md [hunk 32]
```diff
@@ -0,0 +1,7 @@
+
+#### Integrations
+
+##### Rapid7 InsightIDR
+
+- Updating investigation existing commands and adding new investigation commands.
```
Comment (ShirleyDenkberg) [inline Packs/Rapid7_InsightIDR/ReleaseNotes/2_0_0.md:6]:
```suggestion
- Updated investigation existing commands.
- Added new investigation commands.
```
list the new commands that were added.

**Comment by ShirleyDenkberg** [comment]
@JudahSchwartz @sapirshuker Doc review completed.

**Review by JudahSchwartz — APPROVED** [review]

**Pull request merged**